openmc/mesh.py

```diff
@@ -0,0 +1,560 @@
+            # Fortran expects for a real(8), so this avoids code complication
+            # and achieves the same goal.
+            zplanes = [openmc.ZPlane(-1e10, 'reflective'),
+                       openmc.ZPlane(1e10, 'reflective')]
+        else:
+            zplanes = [openmc.ZPlane(self.lower_left[2], bc[4]),
+                       openmc.ZPlane(self.upper_right[2], bc[5])]
+        root_cell = openmc.Cell()
+        root_cell.region = ((+xplanes[0] & -xplanes[1]) &
+                            (+yplanes[0] & -yplanes[1]) &
+                            (+zplanes[0] & -zplanes[1]))
+
+        # Build the universes which will be used for each of the (i,j,k)
+        # locations within the mesh.
+        # We will concurrently build cells to assign to these universes
+        cells = []
+        universes = []
+        for index in self.indices:
+            cells.append(openmc.Cell())
+            universes.append(openmc.Universe())
+            universes[-1].add_cell(cells[-1])
+
+        lattice = openmc.RectLattice()
+        lattice.lower_left = self.lower_left
+
+        # Assign the universe and rotate to match the indexing expected for
+        # the lattice
+        if n_dim == 1:
+            universe_array = np.array([universes])
+        elif n_dim == 2:
+            universe_array = np.empty(self.dimension[::-1],
+                                      dtype=openmc.Universe)
+            i = 0
+            for y in range(self.dimension[1] - 1, -1, -1):
+                for x in range(self.dimension[0]):
+                    universe_array[y][x] = universes[i]
+                    i += 1
+        else:
+            universe_array = np.empty(self.dimension[::-1],
+                                      dtype=openmc.Universe)
+            i = 0
+            for z in range(self.dimension[2]):
+                for y in range(self.dimension[1] - 1, -1, -1):
+                    for x in range(self.dimension[0]):
+                        universe_array[z][y][x] = universes[i]
+                        i += 1
+        lattice.universes = universe_array
+
+        if self.width is not None:
+            lattice.pitch = self.width
+        else:
+            dx = ((self.upper_right[0] - self.lower_left[0]) /
+                  self.dimension[0])
+
+            if n_dim == 1:
+                lattice.pitch = [dx]
+            elif n_dim == 2:
+                dy = ((self.upper_right[1] - self.lower_left[1]) /
+                      self.dimension[1])
+                lattice.pitch = [dx, dy]
+            else:
+                dy = ((self.upper_right[1] - self.lower_left[1]) /
+                      self.dimension[1])
+                dz = ((self.upper_right[2] - self.lower_left[2]) /
+                      self.dimension[2])
+                lattice.pitch = [dx, dy, dz]
+
+        # Fill Cell with the Lattice
+        root_cell.fill = lattice
+
+        return root_cell, cells
+
+
+def Mesh(*args, **kwargs):
```

Comment:
This can also be marked as `@Deprecated` with a decorator.
Reference: https://deprecation.readthedocs.io/en/latest/